### PR TITLE
RUMM-658 Add compatibility with gov cloud

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Allow the user to target the Gov servers.

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/server/LocalServer.kt [hunk 1]
```diff
@@ -37,7 +37,9 @@ class LocalServer {
     }
 
     fun stop() {
-        engine?.stop(SHUTDOWN_MS, STOP_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+        Thread {
```
Comment (mariusc83) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/server/LocalServer.kt:40]: Any specific reason ?

Comment (xgouchet) [reply sample/kotlin/src/main/kotlin/com/datadog/android/sample/server/LocalServer.kt:40]: Yes, the stop method in the server will block the main thread for about a second before returning.

Comment (mariusc83) [reply sample/kotlin/src/main/kotlin/com/datadog/android/sample/server/LocalServer.kt:40]: In this case shouldn't be the case also with the start method ? I am thinking maybe you should also start the server in another Thread. Problem is that doing so you will have to synchronise your code to wait until the server actually starts.

Comment (xgouchet) [reply sample/kotlin/src/main/kotlin/com/datadog/android/sample/server/LocalServer.kt:40]: No cause the start is asynchronous. To be more accurate, stop is not synchronous, it just sends a "stop" signal to the server and waits at least SHUTDOWN_MS and at most STOP_TIMEOUT_MS, effectively blocking the calling thread. 
And no, we don't need to synchronise the code for start/stop, the library handles it safely

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**